src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -244,13 +274,138 @@ TryCreateStypeBoxFromFcinfoAggref(FunctionCallInfo fcinfo)
 	StypeBox *box = pallocInAggContext(fcinfo, sizeof(StypeBox));
 	box->agg = DatumGetObjectId(aggConst->constvalue);
 	HeapTuple aggTuple = GetAggregateForm(box->agg, &aggform);
-	InitializeStypeBox(fcinfo, box, aggTuple, aggform->aggtranstype);
+	InitializeStypeBox(fcinfo, box, aggTuple, aggform->aggtranstype, NULL);
 	ReleaseSysCache(aggTuple);
 
 	return box;
 }
 
 
+/*
+ * CreateAggregationArgumentContext creates an AggregationArgumentContext tailored
+ * to handling the aggregation of input arguments identical to type at
+ * 'argumentIndex' in 'fcinfo'.
+ */
+static AggregationArgumentContext *
+CreateAggregationArgumentContext(FunctionCallInfo fcinfo, int argumentIndex)
+{
+	AggregationArgumentContext *aggregationArgumentContext =
+		pallocInAggContext(fcinfo, sizeof(AggregationArgumentContext));
+
+	/* check if input comes combined into tuple/record */
+	if (RECORDOID == get_fn_expr_argtype(fcinfo->flinfo, argumentIndex))
```

Comment:
Won't this cause an issue if we have a single argument aggregate whose parameter is RECORDOID?

Reply:
I believe such a parameter would not pass our type check, not with this PR and not previously, either.